__tests__/unit/cloudstore.android.test.js

```diff
@@ -0,0 +1,96 @@
+      const setItemSpy = jest.spyOn(GDriveCloudStorage, 'setItem');
+
+      await expect(CloudStore.putKey({ keyId })).rejects.toThrow(/Invalid/);
+      expect(setItemSpy.mock.calls.length).toBe(0);
+    });
+
+    it.skip('store item', async () => {
+      const getItemSpy = jest.spyOn(GDriveCloudStorage, 'getItem').mockImplementation();
+      const setItemSpy = jest.spyOn(GDriveCloudStorage, 'setItem');
+
+      await CloudStore.putKey({ keyId, ciphertext });
+      expect(setItemSpy.mock.calls[0][0]).toBe('1_photon_key_id');
+      expect(setItemSpy.mock.calls[0][1]).toBe(keyId);
+      expect(setItemSpy.mock.calls[1][0]).toBe('1_8abe1a93');
+      expect(setItemSpy.mock.calls[1][1]).toMatch(/^{"keyId":.*"}$/);
+      expect(setItemSpy.mock.calls.length).toBe(2);
+
+      getItemSpy.mockRestore();
+    });
+
+    it.skip('should not backup twice', async () => {
+      const getItemSpy = jest.spyOn(GDriveCloudStorage, 'getItem');
+      const setItemSpy = jest.spyOn(GDriveCloudStorage, 'setItem');
+
+      await expect(CloudStore.putKey({ keyId, ciphertext })).rejects.toThrow(/already present/);
+      expect(setItemSpy.mock.calls.length).toBe(2);
+
+      getItemSpy.mockRestore();
+      setItemSpy.mockRestore();
+    });
+  });
+
+  describe('getKey', () => {
+    it.skip('should not find item', async () => {
+      const getItemSpy = jest.spyOn(GDriveCloudStorage, 'getItem').mockResolvedValue(null);
+
+      const stored = await CloudStore.getKey();
+      expect(stored).toBe(null);
+
+      getItemSpy.mockRestore();
+    });
+
+    it.skip('should get stored item by key Id', async () => {
+      const result = {
+        keyId: keyId,
+        ciphertext: ciphertext,
+        time: new Date(),
+      };
+
+      const getItemSpy = jest
+        .spyOn(GDriveCloudStorage, 'getItem')
+        .mockImplementation()
+        .mockResolvedValueOnce(keyId)
+        .mockResolvedValueOnce(JSON.stringify(result));
+
+      const stored = await CloudStore.getKey();
+      expect(stored).toEqual({
+        keyId,
+        ciphertext,
+        time: expect.objectContaining(new Date()),
+      });
+
+      getItemSpy.mockRestore();
+    });
+
+    describe('removeKeyId', () => {
+      it.skip('fail on invalid args', async () => {
+        const result = {
+          keyId: keyId,
+          ciphertext: ciphertext,
+          time: new Date(),
+        };
+
+        const removeItemSpy = jest.spyOn(GDriveCloudStorage, 'removeItem');
+        const getItemSpy = jest.spyOn(GDriveCloudStorage, 'getItem').mockResolvedValueOnce(keyId).mockResolvedValue(JSON.stringify(result));
+        const getKeySpy = jest.spyOn(CloudStore, 'getKey').mockResolvedValue(result);
+
+        await expect(CloudStore.removeKeyId({ keyId: 'invalid' })).rejects.toThrow(/not found/);
+        expect(getItemSpy.mock.calls.length).toBe(2);
+
+        getItemSpy.mockRestore();
+        removeItemSpy.mockRestore();
+        getKeySpy.mockRestore();
+      });
+    });
+  });
+});
```

Comment:
Nice 👍 Does jest emulate an Android environment if the file is called `*.android.test.js` or how does it work?

Reply:
No unfortunately not, the naming convention just makes it a bit easier to run in isolation [here](https://github.com/photon-sdk/photon-lib/pull/22/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R16) it's not ideal to have a setup file for running in the android environment but changing the Platform directly in the tests doesn't seem to work so this is a workaround using Haste.